src/main/java/org/cactoos/map/MapOf.java

```diff
@@ -148,12 +150,20 @@ public MapOf(final Map<X, Y> src,
      * @param entries List of the entries
      */
     public MapOf(final Iterable<Map.Entry<X, Y>> entries) {
-        super(() -> {
+        this(() -> {
             final Map<X, Y> temp = new HashMap<>(0);
             for (final Map.Entry<X, Y> entry : entries) {
                 temp.put(entry.getKey(), entry.getValue());
             }
             return Collections.unmodifiableMap(temp);
         });
     }
```

Comment:
@andreoss instead of using a Scalar based constructor, I think it would make more sense to follow the same pattern as in `ListOf` and `SetOf`.